packages/pkp-base/src/lib/pkp-base.ts

```diff
@@ -255,6 +254,18 @@ export class PKPBase<T = PKPBaseDefaultParams> {
     }
   }
 
+  public async authenticate(): Promise<void> {
+    this.validateAuthContext();
+
+    const controllerSessionSigs =
+      (await this.authContext?.client?.getSessionSigs(
+        this.authContext.getSessionSigsProps
+      )) || this.controllerSessionSigs;
```

Comment:
This strikes me as a bit odd -- why would we keep the existing value for `this.controllerSessionSigs` here rather than throwing an error if our `authenticate` logic returns a falsy value, or set `this.controllerSessionSigs` to the falsy value that was returned?

Reply:
I was also on the fence about this but wanted a fallback if a previous authenticate call was successful. But probably should just let it throw if not defined after `getSessionSigs` is called.